### PR TITLE
posting a new comment on a article if it exists along with some more …

### PR DESCRIPTION
**Pull request by umerxz**
ENDPOINT: /api/articles/:article_id/comments
METHOD: POST, STATUS: 201
adds a comment for an article.
Request body accepts an object with the following properties:
-username
-body
and responds with the newly posted comment.
ERROR HANDLING
- username does not exist
- article id does not exist
- required field is missing
- invalid type id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -146,4 +146,69 @@ describe("GET /api/articles/:article_id/comments",()=>{
...
+        const newComment = {
+            username: "lurker",
+            body: "I cant code when i am sleepy."
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(newComment)
+        .expect(201)
+        .then(({body})=>{
+            expect(body.comment).toMatchObject({
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:161]: We would want this to be a little more specific as we know what the article id is and author etc

File: __tests__/app.test.js [hunk 2]
```diff
@@ -146,4 +146,69 @@ describe("GET /api/articles/:article_id/comments",()=>{
...
+        }
+        return request(app)
+        .post('/api/articles/1000/comments')
+        .send(newComment)
+        .expect(404)
+        .then((res) => {
+            expect(res.body.msg).toBe("Not Found");
+        });
+    })
+    test('responds with 400 and Bad Request error message when a required field is missing', () => {
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:196]: Lovely work here

File: controllers/articles.controllers.js [hunk 3]
```diff
@@ -24,4 +26,17 @@ exports.getArticleCommentsById = (req, res, next) => {
         res.status(200).send({comments:resolvedPromises[0]})
     })
     .catch(next)
+}
+exports.postArticleCommentById = (req, res, next) => {
+    const article_id=req.params.article_id
+    const {username,body}=req.body
+
+    Promise.all([checkArticleExists(article_id),checkUsersExists(username)])
```
Comment (nc-poppy-m) [inline controllers/articles.controllers.js:34]: Beautiful function! However, article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value. 

If you were to take these checks out, i'd imagine that your PSQL error handlers - provided they're set up correctly - should still catch these.

File: models/articles.models.js [hunk 4]
```diff
@@ -52,4 +52,19 @@ exports.fetchArticleCommentsById = (id) => {
     .then((results) => {
         return (results.rows)
     });
+}
+exports.insertArticleCommentById = (id,username,body) => {
+    if(!id || !username || !body){
+        return Promise.reject({ status:400, msg: "Bad Request"})
+    }
+    const formattedComment = [[Number(id),username,body]]
+    return db.query(
```
Comment (nc-poppy-m) [inline models/articles.models.js:61]: Great work!

**Review by nc-poppy-m — COMMENTED** [review]
Great work! See my comments about additional tests and error handling - you don't need to resubmit the PR for this, but I would suggest implementing the change to your 201 test. You can also refactor to rely on PSQL errors as this would showcase your understanding of them, although this isn't strictly necessary

File: __tests__/app.test.js (hunk 1 above)
Comment (umerxz) [reply __tests__/app.test.js:161]: i was told yesterday in a feedback toMatchObject is more flexible even though i was just getting a single object back (and i was checking for all values in object). So i started using toMatchObject in any test case for objects😬

should i keep switching as per feedback?😂

**Pull request merged**